### ISSUE TITLE
Scan `same-site` for any non-prefix cookies and flag them in production mode checks

### ISSUE THREAD

**Issue by ayumi-cloud** [ayumi-cloud/sc-security-module#460]
### Enhancement idea

- [ ] Scan `same-site` for any non-prefix cookies and flag them in production mode checks.

Extract from 2015:

```
Cookies are a bit of a rough edge on the web's security model. They don't
respect the same origin policy, and the disconnect between their
persistence model and that of the rest of the platform causes no end of
teeth-gnashing and hair-pulling among developers and auditors alike. We're
planning a few changes to bring things into something more closely
resembling alignment:

1.  `secure` ought to mean "secure": Though it's well-known that cookies
cross back and forth over protocol boundaries, recent research
<https://www.usenix.org/system/files/conference/usenixsecurity15/sec15-paper-zheng.pdf>
from
Zheng, et al. has highlighted the fact that this behavior remains
surprising to developers, and also that there's little practical recourse
without changes in the browser.

We intend to make it more difficult for non-secure origins to influence the
state of secure origins by preventing non-secure origins from writing
cookies with a 'secure' flag and overwriting cookies whose 'secure' flag is
set. We're also planning to tweak the eviction rules to ensure that a
host's non-secure cookies are removed before its secure cookies. These
changes are spelled out in more technical detail in an Internet Draft
<https://tools.ietf.org/html/draft-west-leave-secure-cookies-alone> we've
submitted to the IETF for consideration.

The metrics we've gathered show impact to about 0.02% of `Set-Cookie`
operations. Those numbers exclude deletions. Mozilla's metrics
<https://telemetry.mozilla.org/new-pipeline/dist.html#!cumulative=0&end_date=2015-11-19&keys=__none__!__none__!__none__&max_channel_version=release%252F42&measure=COOKIE_SCHEME_SECURITY&min_channel_version=null&product=Firefox!Fennec&sanitize=1&sort_keys=submissions&start_date=2015-10-29&table=0&trim=1&use_submission_date=0>
show similar impact (and include deletions). We don't have metrics

2.  "secureness" should be verifiable from the server-side: We've
implemented Eric Lawrence's clever Cookie Prefixes proposal
<http://textslashplain.com/2015/10/09/duct-tape-and-baling-wirecookie-prefixes/>
as a lightweight mechanism which gives servers the ability to assert
certain properties about cookies they set and receive.

In short, cookies whose names begin with `$Secure-` may only be set from
secure origins, and must have the `secure` flag. Cookies whose names begin
with `$Host-` have the same restrictions, and must additionally have a path
of `/` and no `domain` attribute. This brings them as close to the
same-origin policy as we can get with existing attributes. These changes
are likewise spelled out in more technical detail in the Internet Draft
<https://tools.ietf.org/html/draft-west-cookie-prefixes> we've submitted.
```

Linked to discussion: https://github.com/octobercms/october/issues/5462
